src/branchManagement.ts

```diff
@@ -10,8 +10,22 @@ import { getBranchNameSeparator, getPrefixes, getValidateBranchNameWhiteList } f
  */
 export async function getBranchDetails(): Promise<{ branchNameFinal: string; approval: string } | null> {
     const branchNameSeparator = vscode.workspace.getConfiguration('branch-creator').get<string>('branchNameSeparator', '-');
+    const apps = vscode.workspace.getConfiguration('branch-creator').get<string[]>('appsList', []);
+    const appFirst = vscode.workspace.getConfiguration('branch-creator').get<boolean>('appFirst', false);
+    let appName = '';
+    if (appFirst && apps.length > 0) {
+        appName = await selectAppName();
+        if (!appName) { return null; } // Exit if app selection is required but none is selected
+    }
+
     const prefix = await selectPrefix();
     if (!prefix) { return null; } // Exit if no prefix selected
+
+    if (!appFirst && apps.length > 0) {
```

Comment:
since you're doing this check twice and setting appName wether appFirst exists or doesn't exist I don't think you need to do this if check twice